pydantic/networks.py

```diff
@@ -109,6 +110,8 @@ class AnyUrl(str):
     allowed_schemes: Optional[Set[str]] = None
     tld_required: bool = False
     user_required: bool = False
+    default_parts: Optional[Set[Tuple[str, str]]] = None
```

Comment:
Why not a dictionary? or even a method like
```py
class AnyHttpUrl(AnyUrl):
    allowed_schemes = {'http', 'https'}

    @staticmethod
    def get_default_parts(parts: Dict[str, str]) -> Dict[str, str]:
        return {
            'port': '443' if parts['scheme'] == 'https' else '80',
        }

class HttpUrl(AnyHttpUrl):
    tld_required = True
    # https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers
    max_length = 2083

class RedisDsn(AnyUrl):
    allowed_schemes = {'redis', 'rediss'}

    @staticmethod
    def get_default_parts(parts: Dict[str, str]) -> Dict[str, str]:
        return {
            'domain': 'localhost' if not (parts['ipv4'] or parts['ipv6']) else '',
            'port': '6379',
            'path': '/0',
        }
```

Reply:
Fixed

Reply:
I guess I wanted to use set to have FrozenSet for typing somewhere